common/build/eslint-plugin-fluid/src/rules/no-unchecked-record-access.js

```diff
@@ -303,85 +313,168 @@ function isUndefinableIndexSignatureType(parserServices, node) {
 }
 
 // Helper function to traverse up the code until the scope ends and checks if the property access has been checked for undefined
-function propertyHasBeenChecked(node) {
+function propertyHasBeenChecked(node, context) {
+	const baseObj = getBaseObject(node);
+	const currentKeyNode = node.property;
 	let current = node;
+
 	while (current) {
-		if (isDefined(current)) {
+		if (
+			current.optional ||
+			current.type === "ChainExpression" ||
+			current.type === "TSNonNullExpression"
+		) {
 			return true;
 		}
-		current = current.parent;
-	}
-	return false;
-}
-
-// Helper function to get the type of a node
-function getNodeType(node, parserServices) {
-	const tsNode = parserServices.esTreeNodeToTSNodeMap.get(node);
-	const type = parserServices.program.getTypeChecker().getTypeAtLocation(tsNode);
-	return type;
-}
 
-// Helper function to determine if a node is defined. This has all the cases which define
-function isDefined(node) {
-	if (!node.parent) return false;
-	if (node.optional === true || node.parent.type === "TSNonNullExpression") return true;
+		const parent = current.parent;
+		if (!parent) break;
```

Comment:
Would it be better to explicitly check for undefined (i.e. `if (parent === undefined)`)?